client/src/components/ScrollToTopButton.tsx

```diff
@@ -0,0 +1,33 @@
+import { useState, useEffect } from 'react';
+import { RxTextAlignTop } from 'react-icons/rx';
+
+const ScrollToTopButton = () => {
+  const [showButton, setShowButton] = useState(false);
+
+  useEffect(() => {
+    const handleScroll = () => {
+      setShowButton(window.scrollY > 20); // Show button when scrolled down 200px
+    };
+
+    window.addEventListener('scroll', handleScroll);
+    return () => {
+      window.removeEventListener('scroll', handleScroll);
+    };
+  }, []);
+
+  const scrollToTop = () => {
+    window.scrollTo({ top: 0, behavior: 'smooth' });
+  };
+
+  return (
+    <div className="scroll-to-top text-center">
+      {showButton && (
+        <button onClick={scrollToTop} className="scroll-button">
+          <RxTextAlignTop size={40} />
+        </button>
+      )}
+    </div>
+  );
+};
+
+export default ScrollToTopButton;
```

Comment:
Since you use arrow-function syntax on ScrollToTopButton, you have to use this ugly `export default` style here. That's one reason I dislike arrow-function syntax on React components! ;-)